src/components/Todo.js

```diff
@@ -0,0 +1,50 @@
+import React, { useState } from "react";
+import TodoForm from "./TodoForm";
+import { RiCloseCircleLine } from "react-icons/ri";
+import { TiEdit } from "react-icons/ti";
+
+function Todo({ todos, completeTodo, removeTodo, updateTodo }) {
+  const [editMode, setEditMode] = useState(false);
+  const [edit, setEdit] = useState({
+    id: null,
+    value: "",
+  });
+
+  const submitUpdate = (value) => {
+    updateTodo(value.id, value);
+
+    setEdit({
+      id: null,
+      value: "",
+    });
+  };
+  if (edit.id) {
+    return <TodoForm edit={edit} editMode={editMode} setEditMode={setEditMode} onSubmit={submitUpdate} />;
+  }
+  return todos.map((todo, index) => (
```

Comment:
it is odd to render a list of todos in the `Todo` component. This component should define what a single todo looks like only

Reply:
In the TodoList component I would do something like this:

```
....
return todos.map((todo, index) => <Todo todo={todo} key={index} />
....